pkg/injector/bootstrap.go

```diff
@@ -30,16 +31,16 @@ func (wh *mutatingWebhook) createEnvoyBootstrapFromExisting(newBootstrapSecretNa
 		return nil, fmt.Errorf("error unmarshalling envoy bootstrap config: %w", err)
 	}
 
-	config.Node.Id = cert.GetCommonName().String()
+	config.Node.Id = proxyUUID.String()
 
-	return wh.marshalAndSaveBootstrap(newBootstrapSecretName, namespace, config, cert)
+	return wh.marshalAndSaveBootstrap(bootstrapConfigName(proxyUUID), namespace, config, cert)
 }
 
-func (wh *mutatingWebhook) createEnvoyBootstrapConfig(name, namespace, osmNamespace string, cert *certificate.Certificate, originalHealthProbes models.HealthProbes) (*corev1.Secret, error) {
+func (wh *mutatingWebhook) createEnvoyBootstrapConfig(proxyUUID uuid.UUID, namespace string, cert *certificate.Certificate, originalHealthProbes models.HealthProbes) (*corev1.Secret, error) {
 	builder := bootstrap.Builder{
-		NodeID: cert.GetCommonName().String(),
+		NodeID: proxyUUID.String(),
```

Comment:
Do we need to handle anything on the ADS side for the node id?

Reply:
nope, we never read from it. The current snapshot cache implementation already writes the UUID, so this fixes what was previously broken on it (although it wasn't actually used anyways)